src/bin/cargo/command_prelude.rs

```diff
@@ -359,9 +360,22 @@ pub trait ArgMatchesExt {
                          requires -Zunstable-options to use."
                     ));
                 }
-                Ok(Some(registry.to_string()))
+
+                if registry == "crates.io" {
```

Comment:
I think this may be one of the few places that we accept this string, right? If so, could the special casing not be done here and perhaps later in a more centralized location?

Reply:
Small nit - `crates-io` is used rather than `crates.io` in `[patch]` blocks so it might make sense to use the same here for consistency.

Reply:
> I think this may be one of the few places that we accept this string, right? If so, could the special casing not be done here and perhaps later in a more centralized location?

I have made the string a constant (and updated other uses of it), so it is now just doing a string comparison, I did consider moving the logic into source_id, but we don't always need the source (for example in publish), so I have left it where it was. Is that ok?